website/blog/lineage-from-code/index.mdx

```diff
@@ -0,0 +1,74 @@
+---
+
+# Expanding the Horizon of OpenLineage: Extracting Lineage from Code with Foundational
+
+Data lineage is the cornerstone of modern data governance, providing transparency, traceability, and accountability across the data lifecycle. It ensures that organizations can track how data flows through systems, transforms between processes, and ultimately impacts downstream analytics and decision-making. For data leaders, lineage is critical to maintaining trust in data by ensuring its accuracy, managing risks, and complying with regulatory requirements.
+
+Beyond its role in compliance and security, lineage is essential for operational efficiency. It allows teams to perform impact analyses before making changes, reducing the risk of breaking pipelines or disrupting critical workflows. When combined with metadata and integrated into governance tools, lineage offers a powerful way to visualize data dependencies, troubleshoot issues, and ensure stewardship across the organization.
+
+OpenLineage, the leading standard for capturing and sharing lineage, has transformed how organizations manage runtime-based lineage emitted from tools like Airflow and Spark. Yet, this focus on runtime lineage leaves a gap when it comes to static code, which governs also rarely executed or ad hoc pipelines. Addressing this gap is essential to achieving a comprehensive view of lineage across the data ecosystem.
+
+This blog explores why lineage extracted from code is indispensable, how [Foundational extracts lineage directly from code](https://www.foundational.io/product/data-lineage), the challenges of integrating it into OpenLineage, and how a community-driven approach can address these challenges to provide a holistic view of data lineage.
+
+# The Evolution of Lineage: From Runtime to Code
+
+OpenLineage has excelled at capturing lineage during pipeline execution. Whether tracking Spark transformations or Airflow DAGs, its runtime-centric approach has provided unmatched visibility into active pipelines. However, relying solely on runtime lineage creates blind spots, particularly for rarely executed or complex code paths.
+
+## Examples of Missing Runtime Lineage:
+
+* **Rarely Executed Pipelines**:
+For example, restore jobs triggered only during incidents or annual computations might not emit runtime lineage frequently enough to provide actionable insights.
+* **Specific code-paths that rarely trigger**
+Some pipelines run frequently, but certain code paths are triggered only under rare conditions. For example, dbt incremental models typically run as incremental updates, and full refreshes are seldom executed, leaving their lineage underrepresented.
+* **Rarely Viewed Dashboards**
+Dashboards that are infrequently accessed may not emit queries often enough to appear in runtime lineage. For instance, renaming a column might not show any immediate impact, but it could break an obscure dashboard used only once a quarter or year.
```

Comment:
```suggestion
Dashboards that are infrequently accessed might not emit queries often enough to appear in runtime lineage. For instance, renaming a column might not show any immediate impact, but it could break an obscure dashboard used only once a quarter or year.
```